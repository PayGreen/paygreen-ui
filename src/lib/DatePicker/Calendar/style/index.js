import styled from 'styled-components';

const CalendarBase = styled.div``;

export default CalendarBase;
