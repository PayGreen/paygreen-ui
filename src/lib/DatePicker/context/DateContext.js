import React, { createContext } from 'react';

const DateContext = createContext([null, () => {}]);
const DateContextProvider = ({ value, ...props }) => {
    return (
        <DateContext.Provider value={value}>
            {props.children}
        </DateContext.Provider>
    );
};

export { DateContext, DateContextProvider };
