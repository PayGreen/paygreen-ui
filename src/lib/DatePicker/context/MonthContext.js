import React, { createContext } from 'react';

const MonthContext = createContext([null, () => {}]);
const MonthContextProvider = ({ value, ...props }) => {
    return (
        <MonthContext.Provider value={value}>
            {props.children}
        </MonthContext.Provider>
    );
};

export { MonthContext, MonthContextProvider };
