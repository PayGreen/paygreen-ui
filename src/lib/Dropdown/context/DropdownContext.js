import { createContext, useContext } from 'react';

export const DropdownContext = createContext();

export const useDropdown = () => {
    return useContext(DropdownContext);
};
