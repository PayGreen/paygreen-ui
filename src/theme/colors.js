const mainColors = {
    green: '#33ad73',
    blue: '#00a99d',
    orange: '#e58a28',
    pink: '#ec6a6a',
    purple: '#a26ae1',
    red: '#b54025',
};

const colors = {
    green: {
        main: mainColors.green,
        gradientBase: '#77d1a6',
        gradientShade: '#a4db95',
    },
    blue: {
        main: mainColors.blue,
        gradientBase: '#51c6be',
        gradientShade: '#76bbd8',
    },
    orange: {
        main: mainColors.orange,
        gradientBase: '#fab56b',
        gradientShade: '#faa272',
    },
    pink: {
        main: mainColors.pink,
        gradientBase: '#e98e90',
        gradientShade: '#e98eb9',
    },
    purple: {
        main: mainColors.purple,
        gradientBase: '#b887ee',
        gradientShade: '#dd94ef',
    },
};

export { mainColors, colors };
