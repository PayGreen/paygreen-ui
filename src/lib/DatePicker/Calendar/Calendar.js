import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import {
    formStatusOptions,
    formStatusDefault,
} from '../../../shared/constants';
import { MonthContextProvider } from '../context/MonthContext';
import CalendarNavbar from '../CalendarNavbar/CalendarNavbar';
import CalendarWeekdays from '../CalendarWeekdays/CalendarWeekdays';
import CalendarGrid from '../CalendarGrid/CalendarGrid';

const Calendar = ({
    currentMonth,
    minimumDate,
    maximumDate,
    handleOnChange,
    ...rest
}) => {
    const [month, setMonth] = useState(currentMonth);

    useEffect(() => {
        setMonth(currentMonth);
    }, [currentMonth]);

    return (
        <MonthContextProvider value={[month, setMonth]}>
            <CalendarNavbar {...rest} />
            <CalendarWeekdays {...rest} />
            <CalendarGrid
                minimumDate={minimumDate}
                maximumDate={maximumDate}
                handleOnChange={handleOnChange}
                {...rest}
            />
        </MonthContextProvider>
    );
};

Calendar.propTypes = {
    currentMonth: PropTypes.number,
    minimumDate: PropTypes.object,
    maximumDate: PropTypes.object,
    colorStatus: PropTypes.oneOf(Object.values(formStatusOptions)),
};

Calendar.defaultProps = {
    currentMonth: moment().month(),
    minimumDate: null,
    maximumDate: null,
    colorStatus: formStatusDefault,
};

export default Calendar;
