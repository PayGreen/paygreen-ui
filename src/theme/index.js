import { ThemeDefault } from './theme.default';
import { ThemeDark } from './theme.dark';

export { ThemeDefault, ThemeDark };
