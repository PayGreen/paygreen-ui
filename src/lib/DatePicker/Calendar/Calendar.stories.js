import React, { useState } from 'react';
import moment from 'moment';
import { storiesOf } from '@storybook/react';
import { withKnobs, select, text } from '@storybook/addon-knobs';
import {
    folder,
    formStatusOptions,
    formStatusDefault,
} from '../../../shared/constants';
import Calendar from './Calendar';
import { DateContextProvider } from '../context/DateContext';

storiesOf(folder.form + folder.sub.datePicker + 'Calendar', module)
    .addDecorator(withKnobs)
    .add('Calendar', () => {
        const [selectedDate, setSelectedDate] = useState(moment().startOf('D'));

        return (
            <DateContextProvider value={[selectedDate, setSelectedDate]}>
                <Calendar
                    colorStatus={select(
                        'Color status',
                        formStatusOptions,
                        formStatusDefault,
                    )}
                    minimumDate={moment(
                        text(
                            'Minimum date',
                            moment()
                                .add(-1, 'M')
                                .format('DD/MM/YYYY'),
                        ),
                        'DD/MM/YYYY',
                    )}
                    maximumDate={moment(
                        text(
                            'Maximum date',
                            moment()
                                .add(1, 'M')
                                .format('DD/MM/YYYY'),
                        ),
                        'DD/MM/YYYY',
                    )}
                />
            </DateContextProvider>
        );
    });
