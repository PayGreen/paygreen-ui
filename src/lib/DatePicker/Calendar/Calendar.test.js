import React, { useState } from 'react';
import moment from 'moment';
import TestRenderer from 'react-test-renderer';
import { ThemeDefault } from '../../../theme';
import { DateContextProvider } from '../context/DateContext';
import { DropdownContext } from '../../Dropdown/context/DropdownContext';
import Calendar from './Calendar';

const Calendar_withContext = () => {
    const [isOpen, setOpen] = useState(); // only needed for test
    const [selectedDate, setSelectedDate] = useState(moment());

    return (
        <DropdownContext.Provider value={[setOpen, isOpen]}>
            <DateContextProvider value={[selectedDate, setSelectedDate]}>
                <Calendar theme={ThemeDefault} />
            </DateContextProvider>
        </DropdownContext.Provider>
    );
};

it('renders without crashing', () => {
    const calendar = TestRenderer.create(<Calendar_withContext />);
    expect(calendar.toJSON()).toMatchSnapshot();
});
