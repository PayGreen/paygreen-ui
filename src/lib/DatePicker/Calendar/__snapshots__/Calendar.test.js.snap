// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Array [
  <div
    className="style__CalendarNavbarBase-sc-9kauge-0 cFouVw"
  >
    <button
      className="style__IconBase-ak3679-0 eiqeDU icon"
      onClick={[Function]}
    >
      <svg
        viewBox="0 0 143 143"
      >
        <path
          d="M97.8 123.9c-.8 0-1.6-.3-2.3-.9l-55-47.6c-.8-.7-1.2-1.6-1.2-2.6s.4-2 1.2-2.6l55-48.2c1.5-1.3 3.7-1.1 4.9.3 1.3 1.5 1.1 3.7-.3 4.9l-52 45.6 52 45c1.5 1.3 1.6 3.5.4 4.9-.8.8-1.8 1.2-2.7 1.2z"
        />
      </svg>
    </button>
    January 2021
    <button
      className="style__IconBase-ak3679-0 eiqeDU icon"
      onClick={[Function]}
    >
      <svg
        viewBox="0 0 143 143"
      >
        <path
          d="M43.5 123.9c-1 0-1.9-.4-2.6-1.2-1.3-1.5-1.1-3.7.3-4.9l52-45.6-52-45c-1.5-1.3-1.6-3.5-.4-4.9 1.3-1.5 3.5-1.6 4.9-.4l55 47.6c.8.7 1.2 1.6 1.2 2.6s-.4 2-1.2 2.6l-55 48.2c-.6.7-1.4 1-2.2 1z"
        />
      </svg>
    </button>
  </div>,
  <div
    className="style__CalendarWeekdaysBase-sc-1ai2hag-0 krhkQ"
  >
    <div>
      S
    </div>
    <div>
      M
    </div>
    <div>
      T
    </div>
    <div>
      W
    </div>
    <div>
      T
    </div>
    <div>
      F
    </div>
    <div>
      S
    </div>
  </div>,
  <div
    className="style__CalendarGridBase-sc-1qui6a5-0 ehrtpv"
  >
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      1
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      2
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      3
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      4
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      5
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      6
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      7
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      8
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      9
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      10
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      11
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      12
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      13
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      14
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      15
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      16
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      17
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      18
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 kqyGos"
      onClick={[Function]}
    >
      19
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      20
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      21
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      22
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      23
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      24
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      25
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      26
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      27
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      28
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      29
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      30
    </button>
    <button
      className="style__CalendarCellBase-sc-17uxo6m-0 cEItIt"
      onClick={[Function]}
    >
      31
    </button>
  </div>,
]
`;
